pkg/mcs/tso/server/config.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/pingcap/log"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/tikv/pd/pkg/utils/configutil"
+	"github.com/tikv/pd/pkg/utils/grpcutil"
+	"github.com/tikv/pd/pkg/utils/metricutil"
+	"github.com/tikv/pd/pkg/utils/typeutil"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultLeaderLease   = int64(3)
+	defaultMaxResetTSGap = 24 * time.Hour
+
+	defaultName              = "TSO"
+	defaultBackendEndpoints  = "http://127.0.0.1:2379"
+	defaultListenAddr        = "127.0.0.1:3379"
+	defaultEnableGRPCGateway = true
+
+	defaultTSOSaveInterval           = time.Duration(defaultLeaderLease) * time.Second
+	defaultTSOUpdatePhysicalInterval = 50 * time.Millisecond
+	maxTSOUpdatePhysicalInterval     = 10 * time.Second
+	minTSOUpdatePhysicalInterval     = 1 * time.Millisecond
+
+	defaultLogFormat           = "text"
+	defaultDisableErrorVerbose = true
+)
+
+// NewTSOServiceCommand returns the tso service command.
+func NewTSOServiceCommand() *cobra.Command {
```

Comment:
I suggest keeping it in the main package.

Reply:
I agree

Reply:
The unit test needs to call NewTSOServiceCommand(). Otherwise I need to duplicate the code.

Reply:
to make it code more friendly to test, we'd better move it to stay with the Config.

Reply:
in /Users/binshi/code/pingcap/my-pd/pkg/mcs/tso/server/testutil.go

// NewTestSingleConfig is only for test to create one pd.
// Because PD client also needs this, so export here.
func newTestDefaultConfig() (*Config, error) {
	cmd := NewTSOServiceCommand()
	cfg := NewConfig()
	flagSet := cmd.Flags()
	return cfg, cfg.Parse(flagSet)
}

Reply:
moved back to main. Changed locally.